src/commands/data/import/tree.ts

```diff
@@ -45,23 +46,61 @@ export default class Import extends SfCommand<ImportResult[]> {
     }),
   };
 
-  public async run(): Promise<ImportResult[]> {
+  public async run(): Promise<ImportResult[] | undefined> {
     const { flags } = await this.parse(Import);
 
     const conn = flags['target-org'].getConnection(flags['api-version']);
-    const results = flags.plan
-      ? await importFromPlan(conn, flags.plan)
-      : await importFromFiles(conn, flags.files ?? []);
 
-    this.table({
-      data: results,
-      columns: [
-        { key: 'refId', name: 'Reference ID' },
-        { key: 'type', name: 'Type' },
-        { key: 'id', name: 'ID' },
-      ],
-      title: 'Import Results',
-    });
-    return results;
+    try {
+      const results = flags.plan
+        ? await importFromPlan(conn, flags.plan)
+        : await importFromFiles(conn, flags.files ?? []);
+
+      this.table({
+        data: results,
+        columns: [
+          { key: 'refId', name: 'Reference ID' },
+          { key: 'type', name: 'Type' },
+          { key: 'id', name: 'ID' },
+        ],
+        title: 'Import Results',
+      });
+      return results;
+    } catch (err) {
+      const error = err as SfError;
+      if (
+        error.cause &&
+        error.cause instanceof Error &&
+        'data' in error.cause &&
+        isObject(error.cause.data) &&
+        'message' in error.cause.data
+      ) {
+        const errorData = JSON.parse(error.cause.data.message as string) as TreeResponse;
+
+        if (errorData.hasErrors) {
+          const errorResults = errorData.results
+            .map((result) =>
+              result.errors.map((errors) => ({
+                referenceId: result.referenceId,
+                StatusCode: errors.statusCode,
+                Message: errors.message,
+                fields: errors.fields.join(', ') || 'N/A',
+              }))
+            )
+            .flat();
+
+          this.table({
+            data: errorResults,
+            columns: [
+              { key: 'referenceId', name: 'Reference ID' },
+              { key: 'StatusCode', name: 'Status Code' },
+              { key: 'Message', name: 'Error Message' },
+              { key: 'fields', name: 'Fields' },
+            ],
+            title: 'import Errors',
+          });
```

Comment:
since we manually report the errors in the table we need to manually set the exit code to 1 here:
https://github.com/salesforcecli/plugin-deploy-retrieve/blob/a45867d06f7f511e973a955d4bab164d6486863d/src/commands/project/delete/source.ts#L306